packages/sveltekit/src/server/handle.ts

```diff
@@ -53,30 +70,51 @@ function sendErrorToSentry(e: unknown): unknown {
   return objectifiedErr;
 }
 
-const FETCH_PROXY_SCRIPT = `
+/**
+ * Exported only for testing
+ */
+export const FETCH_PROXY_SCRIPT = `
     const f = window.fetch;
     if(f){
       window._sentryFetchProxy = function(...a){return f(...a)}
       window.fetch = function(...a){return window._sentryFetchProxy(...a)}
     }
 `;
 
-export const transformPageChunk: NonNullable<ResolveOptions['transformPageChunk']> = ({ html }) => {
-  const transaction = getActiveTransaction();
-  if (transaction) {
-    const traceparentData = transaction.toTraceparent();
-    const dynamicSamplingContext = dynamicSamplingContextToSentryBaggageHeader(transaction.getDynamicSamplingContext());
-    const content = `<head>
-  <meta name="sentry-trace" content="${traceparentData}"/>
-  <meta name="baggage" content="${dynamicSamplingContext}"/>
-  <script>${FETCH_PROXY_SCRIPT}
-  </script>
-  `;
-    return html.replace('<head>', content);
-  }
+/**
+ * Adds Sentry tracing <meta> tags to the returned html page.
+ * Adds Sentry fetch proxy script to the returned html page if enabled in options.
+ * Also adds a nonce attribute to the script tag if users specified one for CSP.
+ *
+ * Exported only for testing
+ */
+export function addSentryCodeToPage(options: SentryHandleOptions): NonNullable<ResolveOptions['transformPageChunk']> {
+  const { fetchProxyScriptNonce, injectFetchProxyScript } = options;
+  // if injectFetchProxyScript is not set, we default to true
+  const shouldInjectScript = injectFetchProxyScript !== false;
+  const nonce = fetchProxyScriptNonce ? `nonce="${fetchProxyScriptNonce}"` : '';
 
-  return html;
-};
+  return ({ html }) => {
+    const transaction = getActiveTransaction();
```

Comment:
l: unrelated to this PR, but we should look to rewrite this to use `getActiveSpan()` instead (as transaction will go away). Nothing to do here, just something I noticed :)

Reply:
yeah I had the same thought. Wanted to keep the PR atomic but I'll go over SvelteKit soon to get rid of startTxn and stuff like this.